Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -1964,45 +1971,78 @@ void FilterEvents::generateSplitterTSPalpha(
-      if (0 <= miter->first &&
-          miter->first < static_cast<int>(split_tsp_vec.size())) {
-        DataObjects::EventWorkspace_sptr outws = miter->second;
-        outws->mutableRun().addProperty(std::move(split_tsp_vec[miter->first]),
-                                        true);
-      }
-    }
-  } else {
-    // Either Table-type or Matrix-type splitters
-    for (int itarget = 0; itarget < static_cast<int>(split_tsp_vec.size());
-         ++itarget) {
-      // use itarget to find the workspace that is mapped
-      std::map<int, DataObjects::EventWorkspace_sptr>::iterator ws_iter;
-      ws_iter = m_outputWorkspacesMap.find(itarget);
+  g_log.debug() << "There are " << split_tsp_vec.size()
+                << " TimeSeriesPropeties.\n"
+                << "There are " << m_outputWorkspacesMap.size()
+                << " Output worskpaces.\n";
+
+  if (split_tsp_vec.size() != m_outputWorkspacesMap.size()) {
+    g_log.warning() << "Number of Splitter vector (" << split_tsp_vec.size()
+                    << ") does not match number of output workspace ("
+                    << m_outputWorkspacesMap.size() << ")"
+                    << "\n";
+  }
 
-      // skip if an itarget does not have matched workspace
-      if (ws_iter == m_outputWorkspacesMap.end()) {
-        g_log.warning() << "iTarget " << itarget
-                        << " does not have any workspace associated.\n";
-        continue;
-      }
+  for (int itarget = 0; itarget < static_cast<int>(split_tsp_vec.size());
+       ++itarget) {
+    // use itarget to find the workspace that is mapped
+    std::map<int, DataObjects::EventWorkspace_sptr>::iterator ws_iter;
+    ws_iter = m_outputWorkspacesMap.find(itarget);
 
-      // get the workspace and add property
-      DataObjects::EventWorkspace_sptr outws = ws_iter->second;
-      outws->mutableRun().addProperty(std::move(split_tsp_vec[itarget]), true);
+    // skip if an itarget does not have matched workspace
+    if (ws_iter == m_outputWorkspacesMap.end()) {
+      g_log.warning() << "iTarget " << itarget
+                      << " does not have any workspace associated.\n";
+      continue;
     }
 
-  } // END-IF-ELSE (splitter-type)
+    // get the workspace
+    DataObjects::EventWorkspace_sptr outws = ws_iter->second;
+
+    // calculate the duration
+    double duration = calculate_duration(split_tsp_vec[itarget]);
+
+    // add property
+    PropertyWithValue<double> *duration_property =
+        new PropertyWithValue<double>("duration", duration);
+    outws->mutableRun().addProperty(duration_property, true);
+    // note: split_tps_vec[i], the shared pointer, will be destroyed by
+    // std::move()
+    outws->mutableRun().addProperty(std::move(split_tsp_vec[itarget]), true);
+  }
 
   return;
 }
 
+/** Calculate split-workspace's duration according to splitter time series
+ * property
+ * @brief FilterEvents::calculate_duration
```

Comment:
Remove unhelpful doxygen, or fill in the information

Reply:
Better @brief